docs/faq/programming.md

```diff
@@ -0,0 +1,176 @@
+---
+
+# Venom Blockchain FAQ: Programming smart contracts
+
+<details>
+<summary>
+I know how to develop Ethereum smart contracts. What is the difference with Venom Blockchain?
+</summary>
+
+Your experience with EVM-based blockchains will be useful for developing smart contracts on Venom Blockchain. But there are some differences and the main one is the contracts communication model. You can check [this](../build/development-guides/comparing-of-ethereum-vs-venom-architectures.md) article to know the difference.
+
+Moreover, Venom Blockchain uses Threaded Solidity language, which is a superset of Solidity. You can check its specification [here](https://github.com/tonlabs/TON-Solidity-Compiler/blob/master/API.md)
+
+</details>
+
+<details>
+<summary>
+Which tools can help me for developing on Venom?
+</summary>
+
+Check out [this](../build/quick-start-on-testnet.md) article. We have an awesome [repository](https://github.com/venom-blockchain/awesome-venom) as well.
+
+</details>
+
+<details>
+<summary>
+Does Venom Blockchain have some improvement proposals or some standards for example of fungible/non-fungible tokens?
+</summary>
+
+Sure! You can check the [standards](../standards/readme.md) section, it's right here on the top menu bar :)
+
+</details>
+
+<details>
+<summary>
+How can I get a state of another contract from my contract?
+</summary>
+
+Contracts in Venom can't access any smart contract state directly, because of a special communication model. You can check [this](../build/development-guides/comparing-of-ethereum-vs-venom-architectures.md) article to know the difference. So the only way is to send a message to a contract asking it to return the state you need. The smart contract you are calling should implement a special `responsible` function for such calls. Example:
+
+```solidity
+contract RemoteContract {
+    // Note this function is marked as responsible to call callback function
+    function getCost(uint x) public pure responsible returns (uint) {
+        uint cost = x == 0 ? 111 : 222;
+        // return cost and set option for outbound internal message.
+        return{value: 0, bounce: false, flag: 64} cost;
+    }
+}
+
+contract Caller {
+    function test(address addr, uint x) public pure {
+        // `getCost` returns result to `onGetCost`
+        RemoteContract(addr).getCost{value: 1 ton, callback: Caller.onGetCost}(x);
+    }
+
+    function onGetCost(uint cost) public {
+        // Check if msg.sender is expected address
+        // we get cost value, we can handle this value
+    }
+}
+```
+
+There are many examples of this mechanic within the articles in the [Development Guides](../build/development-guides/readme.md) section. Check this out.
+
+</details>
+
+<details>
+<summary>
+What is the 'flag' of the message and which exactly should I use?
+</summary>
+
+The parameter `flag` of the message determines how much funds will be carried with the message and how to operate with the forward fee.
```

Comment:
The parameter `flag` of the message determines what amount of funds will be carried with the message and how to operate with the forward fee.